### PR TITLE
Add retry when StaleElementReferenceException is raised for Submariner installation

### LINKED ISSUE #9622: Execution fails before ODF installation step with StaleElementReferenceException on running RDR MultiCluster Job
Exception
`E       selenium.common.exceptions.StaleElementReferenceException: Message: stale element reference: stale element not found in the current frame
2024-04-03 12:48:52  E         (Session info: chrome-headless-shell=123.0.6312.105)
2024-04-03 12:48:52  
2024-04-03 12:48:52  venv/lib64/python3.8/site-packages/selenium/webdriver/remote/errorhandler.py:242: StaleElementReferenceException`

Job details - https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-rdr-setup/658/console

### PR DESCRIPTION
**Pull request by am-agrawa**
fixes #9622

### CODE REVIEW

File: ocs_ci/ocs/acm/acm.py [hunk 1]
```diff
@@ -233,8 +236,18 @@ def install_submariner_ui(self, globalnet=True):
...
+            try:
+                WebDriverWait(self, 10).until(
+                    ec.presence_of_element_located(
+                        self.do_click(self.page_nav["install-submariner-btn"])
+                    )
+                )
+                log.info("Click on 'Install Submariner add-ons' button")
+            except StaleElementReferenceException:
+                if attempt == retries - 1:
+                    raise
```
[the remark this answers is not in the document]
Comment (am-agrawa) [reply ocs_ci/ocs/acm/acm.py:254]: done

**Review by ebondare — DISMISSED** [review]

File: ocs_ci/ocs/acm/acm.py [hunk 2]
```diff
@@ -233,8 +238,25 @@ def install_submariner_ui(self, globalnet=True):
         time.sleep(3)
         log.info("Click on 'Submariner add-ons' tab")
         self.do_click(self.page_nav["submariner-tab"])
-        log.info("Click on 'Install Submariner add-ons' button")
-        self.do_click(self.page_nav["install-submariner-btn"], timeout=120)
+        retries = 6
+        for attempt in range(retries):
+            try:
+                WebDriverWait(self, 10).until(
```
Comment (DanielOsypenko) [inline ocs_ci/ocs/acm/acm.py:244]: here you need to pass self.driver, but not self to constructor of WebDriverWait

File: ocs_ci/ocs/acm/acm.py [hunk 3]
```diff
@@ -234,7 +240,9 @@ def install_submariner_ui(self, globalnet=True):
         log.info("Click on 'Submariner add-ons' tab")
         self.do_click(self.page_nav["submariner-tab"])
         log.info("Click on 'Install Submariner add-ons' button")
-        self.do_click(self.page_nav["install-submariner-btn"], timeout=120)
+        retry((StaleElementReferenceException, TimeoutException), retries=5, delay=10)(
```
Comment (DanielOsypenko) [inline ocs_ci/ocs/acm/acm.py:243]: if you haven't tested it yet, pls consider the first option we discussed, it refreshes the page in between retries.
```
 self.do_click(
            self.page_nav["install-submariner-btn"], enable_screenshot=True, avoid_stale=True
        )
```

**Comment by am-agrawa** [comment]
Fixed as part of pr#https://github.com/red-hat-storage/ocs-ci/pull/9646

File: ocs_ci/ocs/acm/acm.py (hunk 3 above)
Comment (am-agrawa) [reply ocs_ci/ocs/acm/acm.py:243]: done in #9646

**Pull request closed without merging**